streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -554,6 +561,10 @@ private void runLoop() {
                                  "trigger a new rebalance.");
                     assignmentErrorCode.set(AssignorError.NONE.code());
                     mainConsumer.enforceRebalance();
+                } else if (nextProbingRebalanceMs.get() < System.currentTimeMillis()) {
```

Comment:
```suggestion
                } else if (nextProbingRebalanceMs.get() < Time.milliseconds()) {
```

This will let us control it in testing, which we should do.